### ISSUE TITLE
LOG macro conflicts

### ISSUE THREAD

**Issue by kyessenov** [proxy-wasm/proxy-wasm-cpp-sdk#154]
Abseil has defined LOG as well: https://github.com/abseil/abseil-cpp/blob/master/absl/log/log.h

**Comment by mpwarres** [comment]
That's an unfortunate collision, though I guess it isn't surprising. We can't simply switch to something like `PROXY_WASM_LOG` since that would break existing code. But maybe we could have a conditional define similar to [`PROXY_WASM_PROTOBUF`](https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/blob/921039ae983ce053bf5cba78a85a3c08ff9791e5/proxy_wasm_api.h#L24), which (if defined) would cause `proxy_wasm_api.h` to define the `LOG` macros as `PROXY_WASM_LOG` instead of `LOG`.

@kyessenov supposing we added such a `USE_PROXY_WASM_LOG` macro, would that address your issue?

**Comment by kyessenov** [comment]
Yeah, that would work. The problem is mainly "null VM" code since it imports both parts of Envoy with absl and proxy-wasm header. I think it's fine to disable the log macros in those modules since they are not standalone and coupled with Envoy.

**Comment by PiotrSikora** [comment]
Does anybody use `LOG` directly? I think it's an internal implementation detail and end users use `LOG_TRACE`, etc. If that's the case, then we should be able to rename it without any extra defines.
